cmd/create/cluster/cmd.go

```diff
@@ -3017,6 +3077,55 @@ func run(cmd *cobra.Command, _ []string) {
 	}
 }
 
+func GetBillingAccountContracts(cloudAccounts []*accountsv1.CloudAccount,
+	billingAccount string) ([]*accountsv1.Contract, bool) {
+	var contracts []*accountsv1.Contract
+	var isContractEnabled bool
+	for _, account := range cloudAccounts {
+		if account.CloudAccountID() == billingAccount {
+			contracts = account.Contracts()
+			for _, contract := range contracts {
+				for _, dimension := range contract.Dimensions() {
+					if ocm.HasValidContracts(dimension) {
+						isContractEnabled = true
+						break
+					}
+				}
+			}
+
+		}
+	}
+	return contracts, isContractEnabled
+}
+
+func PrintContracts(contracts []*accountsv1.Contract, r *rosa.Runtime) {
+	for _, contract := range contracts {
+		format := "Jan 02, 2006"
+		dimensions := contract.Dimensions()
+		var prePurchaseInfo string
+
+		for _, dimension := range dimensions {
+			numberOfVPCs := ocm.GetDimensionValue(dimension, "four_vcpu_hour")
+			numberOfClusters := ocm.GetDimensionValue(dimension, "control_plane")
```

Comment:
@chenz4027 I believe this logic needs to be adjusted. My understanding is that it's not implementing the behaviour you would expect, i.e: the dimensions are not correctly assigned to the vars. PTAL

Reply:
@cristianoveiga I'm a bit lost on this one, would you mind elaborating a bit more on this?

Reply:
@chenz4027 I think what @cristianoveiga is saying is you might overwrite your previously set value with a 0 with the current logic.

So let say dimensions is [ 4 vpcs , 2 clusters ]
The first time it goes through the loop 
numberOfVPCs get set to 4
numberOfClusters get set to 0
then second time it goes through the loop
numberOfVPCs is now set to 0 (!?)
numberOfClusters is now set to 2.

I think what you need to do to fix it is use a if or case statements to only set numberOfVPCs when the dimension name matches four_vcpu_hour and only set numberOfClusters when dimension matches control_plane

Reply:
New implementation: https://github.com/openshift/rosa/pull/1500/files#diff-54e259e8c0b289af4e61624a5e697f0285de9d9e0e14375c7d4fa82c772038eaR86

Reply:
IMO, this is still a bit off. The main point is that you are calling GetDimensionValue inside the dimensions loop. This way will either get a valid value for the four_vcpu_hour or control_plane.

Reply:
Updated